astronomer/providers/amazon/aws/example_dags/example_redshift_sql.py

```diff
@@ -54,17 +57,24 @@ def delete_redshift_cluster() -> None:
             SkipFinalClusterSnapshot=True,
         )
 
-        while True:
-            if get_cluster_status() == "deleting":
-                time.sleep(30)
-                continue
+        while get_cluster_status() == "deleting":
+            time.sleep(30)
+
     except ClientError as exception:
-        logging.exception("Error deleting redshift cluster")
-        raise exception
+        if exception.response.get("Error", {}).get("Code", "") == "ClusterNotFound":
```

Comment:
Should we add a check on L55 to see if the cluster exists or not? try..except is good too :) but just thinking out loud

Reply:
yes, like the idea of being proactively checking rather than reacting with a try-catch. We have the except more for the while loop on L60 where the cluster remains in deleting state for a while and then throws this error. In my opinion it makes sense for L55 to have that check as it will be only once; but for L60 it will then make 2 API calls each time in the loop. 
However, I do not seem to find a relevant method available to check beforehand that the cluster exists https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/redshift.html